akka-http/src/main/scala/akka/http/scaladsl/server/directives/SecurityDirectives.scala

```diff
@@ -254,7 +256,14 @@ trait SecurityDirectives {
    *
    * @group security
    */
-  def challengeFor(realm: String) = HttpChallenge(scheme = "Basic", realm = realm, params = Map.empty)
+  def basicChallengeFor(realm: String) = HttpChallenge(scheme = "Basic", realm = realm, params = Map.empty)
```

Comment:
Let's keep it simple and remove these methods (scala + java). Adding tons of methods with default vals doesn't scale and they aren't reused anywhere else anyway.